rlpx/rlpx.go

```diff
@@ -0,0 +1,439 @@
+// Implements the RLPx protocol. Code is based off the devp2p spec defined here:
+// https://github.com/ethereum/devp2p/blob/master/rlpx.md
+package rlpx
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"hash"
+	mrand "math/rand"
+	"net"
+
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"golang.org/x/crypto/sha3"
+
+	"github.com/indexsupply/x/bint"
+	"github.com/indexsupply/x/ecies"
+	"github.com/indexsupply/x/enr"
+	"github.com/indexsupply/x/isxerrors"
+	"github.com/indexsupply/x/isxhash"
+	"github.com/indexsupply/x/isxsecp256k1"
+	"github.com/indexsupply/x/rlp"
+)
+
+type session struct {
+	Verbose bool
+
+	conn   net.Conn
+	local  *enr.Record
+	ig, eg *mstate
+}
+
+func (s *session) log(format string, args ...any) {
+	if s.Verbose {
+		fmt.Printf(format, args...)
+	}
+}
+
+func Session(l *enr.Record, hs *Handshake) (*session, error) {
+	var (
+		s   = &session{local: l}
+		err error
+	)
```

Comment:
Might be good idea to throw an error here if the Handshake isn't complete (ie missing both nonces, ephemeral keys, pub keys, etc).

Reply:
Good call!

Reply:
fixed!